### PR TITLE
Set-up folder for the Home GW and add relevant documentation

### PR DESCRIPTION
**Pull request by gvancuts**
The folder structure was not very logical anymore and it was the result
of a variety of additional components that have been added over the past
few months. Some parts of the documentation (README.md) was out-of-date
or non-existent for some parts. This commit separates the various pieces
in a more logical manner and adds some (at times basic) documentation
via various README.md files located in the various folders.

With one minor exception, no changes to the actual code was made (not
needed). The only exception to that is the path used in a service file.

Signed-off-by: Geoffroy Van Cutsem <geoffroy.vancutsem@intel.com>

### CODE REVIEW

**Comment by gvancuts** [comment]
@ttzeng, this invalidates the instructions on https://01.org/smarthome in a couple of places:
1. Where we indicate how to start first_server.js on the gateway
2. Where we describe how to build a Docker image for the cloud portal

All others, PTAL

**Comment by gvancuts** [comment]
@nagineni upon inspecting the first_server.js code and other files, I came to the conclusion that they had been written in a relocatable manner (using ./ for a "require" and __dirname elsewhere)... I have not actually tested this however so if you or @ttzeng could test it out, it would be much appreciated.

**Comment by nagineni** [comment]
Also re-naming first_server.js to something more descriptive make sense IMO.

**Comment by gvancuts** [comment]
Thanks for the comment @nagineni , it makes sense to me. I'll make this changes and update the PR. Any idea/suggestion on the new name for first_server.js? (home_gateway_server.js, ...)

**Comment by nagineni** [comment]
`home_gateway_server.js` or just `gateway_server.js`

**Comment by gvancuts** [comment]
I've renamed the home-gateway folder into gateway (making it more generic as it is not specific to a SmartHome scenario). The first_server.js has been renamed into gateway-server.js

@nagineni PTAL

**Comment by nagineni** [comment]
:+1:

@gvancuts, thanks!

**Comment by gvancuts** [comment]
Thanks @nagineni 

@ichbinblau @ttzeng , I want to make sure you are both OK with this change before merging. No rush but please take a look and let me know one way of the other if you agree with this PR.

**Comment by ichbinblau** [comment]
Hi, @gvancuts I am good with the changes. The new folder naming makes more sense.

File: README.md [hunk 1]
```diff
@@ -1,50 +1,12 @@
-# SmartHome "Dollhouse" demo
+# The IoT to Cloud SmartHome demo
 
-This repository provides source code that allows to set-up and demonstrate a SmartHome scenario leveraging [OCF](http://openconnectivity.org/) and [IoTivity](https://www.iotivity.org/) to communicate between smart appliances (OCF servers) and a Home GW (OCF client). The User Interface (UI) is a 3D model of the house (<insert reference here>) that is exposed through a webserver running on the HomeGW.
+This repository provides source code that allows to set-up and demonstrate an IoT to Cloud SmartHome demo leveraging [OCF](http://openconnectivity.org/) and [IoTivity](https://www.iotivity.org/) to communicate between smart appliances (OCF servers), a Home GW (OCF client) and a cloud-based portal (using the [IoT REST API Server](https://github.com/01org/iot-rest-api-server/blob/master/README.md)). The user can interact with the various devices via both an Android companion app (when on the local network) and via the cloud portal. For a more comprehensive overview of the architecture and a detailed step-by-step guide on how to put it all together, please take a look at our online [tutorial](https://01.org/smarthome).
```
Comment (ttzeng) [inline README.md]: The link of 'IoT REST API Server' points to the blob of README.md file of the project, suggest to point to the guithub [home](https://github.com/01org/iot-rest-api-server) of the project.

File: gateway/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# The Home Gateway
+
+The demo was built leveraging the [Ostro OS
+Project](https://ostroproject.org/) and we recommend you use it to build your
```
Comment (ttzeng) [inline gateway/README.md]: Are we still recommending Ostro? suggest to rephrase or remove words till next sentence.

File: gateway/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+own demo.
+Ostro is a Linux operating system tailored for IoT
+smart devices and it fulfills many of the software stack requirements to run
+this demo out of the box.
+
+## Node.js dependencies
+
+Node.js dependencies can be installed using `npm install <node_module>` (you need a live network connection) for those not already provided by the OS.
+* [Express](https://www.npmjs.com/package/express)
+* [http](https://www.npmjs.com/package/http)
```
Comment (ttzeng) [inline gateway/README.md]: it seems 'http' module is no longer required.

File: README.md [hunk 4]
```diff
@@ -1,50 +1,12 @@
...
 
 The repository is organised as follows:
-* root: the root folder is the code code running on the HomeGW
-* `ocf-servers`: provides a number of OCF server implementations in JavaScript (located in `js-servers`). The documentation for these servers is availabe in the `doc/` folder. These are not strictly part of the code running on the gateway but are provided here as a convenience.
+* `gateway`: code running on the Home Gateway
+* `ocf-servers`: OCF server implementations in JavaScript (located in `js-servers`). The documentation for these servers is availabe in the `ocf-servers/doc/` folder.
```
Comment (ttzeng) [inline README.md]: The 'doc' sub-folder actually contains the spec of the supported OCF servers, the documentation README.md of the servers is located in the 'js-servers' folder.

File: ostro-config/systemd-services/first-server/smarthome.service [hunk 5]
```diff
@@ -3,7 +3,7 @@ Description=SmartHome demo startup service
 After=network.target
```
Comment (ttzeng) [inline ostro-config/systemd-services/first-server/smarthome.service]: 'first-server' still contains in the directory path SmartHome-Demo/ostro-config/systemd-services**/first-server**/smarthome.service

**Comment by ttzeng** [comment]
In additions above inspections, this PR (with [commit c183037](https://github.com/01org/SmartHome-Demo/tree/c183037cee39f072a96f52327da9925046c52cbe)) has been validated on real devices, the motion, fan, buzzer, RGB LED, solar servers on Edison can be controlled by the companion app and the 3D home UI.

**Comment by ttzeng** [comment]
Submitted the change requests to Doug to update the tutorial as well.

File: README.md (hunk 1 above)
Comment (gvancuts) [inline README.md]: Makes sense, I will update the PR accrodingly

File: gateway/README.md (hunk 2 above)
Comment (gvancuts) [inline gateway/README.md]: It's still what we are using in practice and what's documented in the tutorial, I'd prefer to keep it in here until we have an alternative properly documented.

File: gateway/README.md (hunk 3 above)
Comment (gvancuts) [inline gateway/README.md]: Noted, I will remove it

File: ostro-config/systemd-services/first-server/smarthome.service (hunk 5 above)
Comment (gvancuts) [inline ostro-config/systemd-services/first-server/smarthome.service]: Good catch... I had grepped for first_server only and missed this... I will update this path too!

**Comment by gvancuts** [comment]
I have force-pushed an updated PR. Let me know if you have additional comments @ttzeng

**Comment by ttzeng** [comment]
I have no further comment. (y)

**Pull request merged**